### PR TITLE
add status roles

### PR DESCRIPTION
**Pull request by Thiblef**

### CODE REVIEW

File: ApiBundle/Transformer/StatusTransformer.php [hunk 1]
```diff
@@ -34,12 +36,14 @@ public function __construct(
...
         TranslationChoiceManager $translationChoiceManager,
-        TranslatorInterface $translator
+        TranslatorInterface $translator,
+        AuthorizationCheckerInterface $authorizationChecker
     ) {
         $this->authorizeStatusChangeManager = $authorizeStatusChangeManager;
         $this->roleRepository = $roleRepository;
         $this->translationChoiceManager = $translationChoiceManager;
         $this->translator = $translator;
+        parent::__construct($authorizationChecker);
```
Comment (itkg-alavieille) [inline ApiBundle/Transformer/StatusTransformer.php]: call parent constructor before set parameters

**Comment by Thiblef** [comment]
@itkg-alavieille

File: ApiBundle/Transformer/StatusTransformer.php [hunk 2]
```diff
@@ -81,14 +85,19 @@ public function transform($status, $document = null)
...
-                array('statusId' => $status->getId())
-            ));
+            if ($this->authorizationChecker->isGranted(AdministrationPanelStrategy::ROLE_ACCESS_DELETE_STATUS)) {
+                $facade->addLink('_self_delete', $this->generateRoute(
+                    'open_orchestra_api_status_delete',
+                    array('statusId' => $status->getId())
+                ));
+            }
+
+            if($this->authorizationChecker->isGranted(AdministrationPanelStrategy::ROLE_ACCESS_UPDATE_STATUS)) {
```
Comment (itkg-alavieille) [inline ApiBundle/Transformer/StatusTransformer.php]: psr2

**Comment by itkg-alavieille** [comment]
else psr2 :+1:

**Pull request merged**